services/placement/service/placementservice_test.go

```diff
@@ -702,3 +718,27 @@ func (ms *mockStorage) Placement(service services.ServiceID) (services.ServicePl
 
 	return nil, 0, errors.New("placement not exist")
 }
+
+func markAllInstancesAvailable(
+	t *testing.T,
+	ps services.PlacementService,
+) {
+	p, err := ps.Placement()
+	require.NoError(t, err)
+	for _, i := range p.Instances() {
+		if len(i.Shards().ShardsForState(shard.Initializing)) == 0 {
+			continue
+		}
+		err := ps.MarkInstanceAvailable(i.ID())
+		require.NoError(t, err)
+	}
+}
+
+func placementToString(
```

Comment:
It's useful to have around, I'd probably end up re-writing it next time I come back to these tests.  Perhaps it'd be better to put it as a `String()` method on `ServicePlacement` however.

I might do that unless you have objections?